include/databroker/participant/ParticipantDatabase.hpp

```diff
@@ -22,12 +22,31 @@
 #include <map>
 
 #include <databroker/types/ParticipantId.hpp>
-#include <databroker/writer/IDatabrokerWriter.hpp>
+#include <databroker/participant/IDatabrokerParticipant.hpp>
 
 namespace eprosima {
 namespace databroker {
 
-using ParticipantDatabase = std::map<ParticipantId, std::shared_ptr<IDatabrokerParticipant>>
+class ParticipantDatabase
+{
+public:
+
+    ParticipantDatabase() = default;
+
+    virtual ~ParticipantDatabase();
+
+    // WARNING only used by Databroker
+    std::shared_ptr<IDatabrokerParticipant> add_participant(
```

Comment:
I think that this function should return `void` as you are returning the same pointer passed as argument

Reply:
:+1:  (typo)